canal/canal.go

```diff
@@ -185,6 +185,12 @@ func (c *Canal) StartFromGTID(set mysql.GTIDSet) error {
 	return c.Run()
 }
 
+// Dump all data from MySQL master `mysqldump`, ignore sync binlog.
+func (c *Canal) Dump() error {
```

Comment:
seem that if we call this function we can't call Run at all, maybe we should add some comments here.